test/org/sagebionetworks/bridge/hibernate/HibernateAccountDaoTest.java

```diff
@@ -850,13 +851,72 @@ public void getByEmailNotFound() {
+        // mock hibernate
+        when(mockHibernateHelper.queryGet("from HibernateAccount where studyId='"
+                + TestConstants.TEST_STUDY_IDENTIFIER + "' and phone='" + TestConstants.PHONE.getNumber() + 
+                "'", null, null, HibernateAccount.class)).thenReturn(ImmutableList.of(hibernateAccount));
 
+        // execute and validate
+        Account account = dao.getAccount(ACCOUNT_ID_WITH_PHONE);
+        assertEquals(hibernateAccount.getEmail(), account.getEmail());
+    }
+    
     @Test
-    public void delete() {
-        dao.deleteAccount(STUDY, ACCOUNT_ID);
+    public void getByPhoneNotFound() {
+        when(mockHibernateHelper.queryGet("from HibernateAccount where studyId='"
+                + TestConstants.TEST_STUDY_IDENTIFIER + "' and phone='" + TestConstants.PHONE.getNumber() + 
+                "'", null, null, HibernateAccount.class)).thenReturn(ImmutableList.of());
+        
+        Account account = dao.getAccount(ACCOUNT_ID_WITH_PHONE);
+        assertNull(account);
+    }
+
+    @Test
+    public void getByPhoneAfterAuthentication() throws Exception {
+        HibernateAccount hibernateAccount = makeValidHibernateAccount(false, false);
+        // mock hibernate
+        when(mockHibernateHelper.queryGet("from HibernateAccount where studyId='"
+                + TestConstants.TEST_STUDY_IDENTIFIER + "' and phone='" + TestConstants.PHONE.getNumber() + 
+                "'", null, null, HibernateAccount.class)).thenReturn(ImmutableList.of(hibernateAccount));
+
+        // execute and validate
+        Account account = dao.getAccountAfterAuthentication(ACCOUNT_ID_WITH_PHONE);
+        assertEquals(hibernateAccount.getEmail(), account.getEmail());
+    }
+    
+    @Test
+    public void getByPhoneNotFoundAfterAuthentication() {
+        when(mockHibernateHelper.queryGet("from HibernateAccount where studyId='"
+                + TestConstants.TEST_STUDY_IDENTIFIER + "' and phone='" + TestConstants.PHONE.getNumber() + 
+                "'", null, null, HibernateAccount.class)).thenReturn(ImmutableList.of());
+        
+        Account account = dao.getAccountAfterAuthentication(ACCOUNT_ID_WITH_PHONE);
+        assertNull(account);
+    }
+    
+    @Test
+    public void deleteWithoutId() throws Exception {
+        // Can't use email, so it will do a lookup of the account
+        HibernateAccount hibernateAccount = makeValidHibernateAccount(false, false);
+        when(mockHibernateHelper.queryGet(any(), any(), any(), any())).thenReturn(ImmutableList.of(hibernateAccount));
+        
+        dao.deleteAccount(ACCOUNT_ID_WITH_EMAIL);
+        
+        verify(mockHibernateHelper).deleteById(HibernateAccount.class, ACCOUNT_ID);
+    }
+    
+    @Test
+    public void deleteWithId() throws Exception {
+        // Directly deletes with the ID it has
+        dao.deleteAccount(ACCOUNT_ID_WITH_ID);
+        
         verify(mockHibernateHelper).deleteById(HibernateAccount.class, ACCOUNT_ID);
```

Comment:
Is it worth adding a check that we *don't* re-query Hibernate for the account?

Reply:
Added